image/v1/types.go

```diff
@@ -49,8 +49,8 @@ type Image struct {
 	DockerImageMetadataVersion string `json:"dockerImageMetadataVersion,omitempty" protobuf:"bytes,4,opt,name=dockerImageMetadataVersion"`
 	// DockerImageManifest is the raw JSON of the manifest
 	DockerImageManifest string `json:"dockerImageManifest,omitempty" protobuf:"bytes,5,opt,name=dockerImageManifest"`
-	// DockerImageLayers represents the layers in the image. May not be set if the image does not define that data.
-	DockerImageLayers []ImageLayer `json:"dockerImageLayers" protobuf:"bytes,6,rep,name=dockerImageLayers"`
+	// DockerImageLayers represents the layers in the image. May not be set if the image does not define that data or if the image represents a manifest list.
+	DockerImageLayers []ImageLayer `json:"dockerImageLayers,omitempty" protobuf:"bytes,6,rep,name=dockerImageLayers"`
```

Comment:
Is this required or optional do you know?

I don't see a generated CRD either, do you not have a CRD?

Reply:
If the image represents a regular manifest then this field is required; if it represents a manifest list then this field makes no sense and should not be present.

Reply:
and there's no CRD, these objects are native to our api server.

Reply:
> If the image represents a regular manifest then this field is required; if it represents a manifest list then this field makes no sense and should not be present.

This implies to me that we should probably be using a discrminated union, though, you have other fields that should be in an union which might make this even more confusing in the future , ref https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions

> and there's no CRD, these objects are native to our api server.

Ok cool, are you doing your own validation manually rather than using openapi validations then?

Reply:
> This implies to me that we should probably be using a discriminated union, though, you have other fields that should be in an union which might make this even more confusing in the future , ref https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions

This field is populated by our own code and not by the user. I'm not sure how much value using a discriminated union would add 🤔 

> Ok cool, are you doing your own validation manually rather than using openapi validations then?

Not sure how validation works for these, I'll investigate and get back to you.

Reply:
> This field is populated by our own code and not by the user. I'm not sure how much value using a discriminated union would add 🤔

Is it a status field or a spec field consumed by another operator?

Reply:
It's not spec nor status. Actually this object doesn't have spec/status fields. It more like a read-only config map. It is used primarily by the image registry and the console.